src/ServiceControl/Licensing/LicenseCheckFeature.cs

```diff
@@ -0,0 +1,58 @@
+﻿namespace Particular.ServiceControl.Licensing
+{
+    using System;
+    using System.Threading;
+    using NServiceBus;
+    using NServiceBus.Features;
+    
+    class LicenseCheckFeature : Feature
+    {
+        public LicenseCheckFeature()
+        {
+            EnableByDefault();
+            RegisterStartupTask<LicenseCheckFeatureStartup>();
+        }
+        
+        protected override void Setup(FeatureConfigurationContext context)
+        {
+            context.Container.ConfigureComponent<ActiveLicense>(DependencyLifecycle.SingleInstance);
+        }
+    }
+
+    class LicenseCheckFeatureStartup : FeatureStartupTask
+    {
+        ActiveLicense activeLicense;
+        Timer timer;
+
+        public LicenseCheckFeatureStartup(ActiveLicense activeLicense)
+        {
+            this.activeLicense = activeLicense;
+        }
+
+        protected override void OnStart()
+        {
+            timer = new Timer(Refresh, null, (int)TimeSpan.FromMinutes(1).TotalMilliseconds, -1);
+        }
+
+        protected override void OnStop()
+        {
+            using (var manualResetEvent = new ManualResetEvent(false))
+            {
+                timer.Dispose(manualResetEvent);
+                manualResetEvent.WaitOne();
+            }
+        }
+
+        void Refresh(object _)
+        {
+            activeLicense.Refresh();
+            try
+            {
+                timer.Change((int)TimeSpan.FromMinutes(10).TotalMilliseconds, -1);
```

Comment:
I just had a look at the Raven code - it checks for a license every 15 minutes.  An hour seems like a long time.  If we're going to set it to be that long then I'll change the mgmt UI to indicate that when updating/adding a license